### PR TITLE
Add header search

### PR DESCRIPTION
**Pull request by huwshimi**
Add the search interface to the header.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- visit `http://0.0.0.0:6543/:flags:/react/`
- click on "Search the store"
- the search box should expand and the background should go white
- click on the canvas
- the search box should contract and the background should return to grey
- click on the search box
- type something and hit enter or click the search icon
- the search results should appear
- the search box should stay expanded

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3033/
Test PASSed.

File: jujugui/static/gui/src/app/components/header-search/header-search.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+              onClick={this._handleSubmit}
+              className="header-search__submit"
+              dangerouslySetInnerHTML={{__html: this.icons.submit}} />
+            <input type="search" name="query"
+              className="header-search__input"
+              placeholder="Search the store"
+              onFocus={this._handleSearchFocus}
+              onBlur={this._handleSearchBlur}
+              style={this.state.inputStyles}
+              valueLink={this.linkState('query')} />
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/header-search/header-search.js]: Is this really any better? I'm a little worried those who aren't in the know will not understand what the mixin does.

**Comment by hatched** [comment]
:+1: QA OK! 

This is awesome, good work!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3038/
Test PASSed.

**Comment by huwshimi** [comment]
Thanks :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**